signature-aggregator/api/api.go

```diff
@@ -0,0 +1,154 @@
+// Copyright (C) 2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package api
+
+import (
+	"encoding/hex"
+	"encoding/json"
+	"net/http"
+	"strings"
+
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/utils/logging"
+	"github.com/ava-labs/awm-relayer/signature-aggregator/aggregator"
+	"github.com/ava-labs/awm-relayer/types"
+	"github.com/ava-labs/awm-relayer/utils"
+	"go.uber.org/zap"
+)
+
+const (
+	APIPath          = "/aggregate-signatures"
+	QuorumPercentage = 67
+)
+
+// Defines a request interface for signature aggregation for a raw unsigned message.
+type AggregateSignatureRequest struct {
+	// Required. hex-encoded message, optionally prefixed with "0x".
+	UnsignedMessage string `json:"unsigned-message"`
+	// Optional hex or cb58 encoded signing subnet ID. If omitted will default to the subnetID of the source blockchain
+	SigningSubnetID string `json:"signing-subnet-id"`
+	// Optional. Integer from 0 to 100 representing the percentage of the quorum that is required to sign the message
+	// defaults to 67 if omitted.
+	QuorumPercentage uint64 `json:"quorum-percentage"`
+}
+
+type AggregateSignatureResponse struct {
+	// hex encoding of the signature
+	SignedMessage string `json:"signed-message"`
+}
```

Comment:
We should consider returning the P-Chain height corresponding to the signature's canonical validator bit vector. Without it, there's an implicit "expiration" on responses from the API. With it, the caller would be better able to implement checks against the current state of the P-Chain before sending a tx to deliver the Warp message.

Reply:
I'm happy to add this to follow up. Since we are already changing interfaces there for ACP-118. But I do think that the implicit P-chain height is current height and the general workflow is request it and send it immediately. Failure in that case is extremely unlikely but a simple retry of the full flow should be an easy fix.